clang/test/Analysis/out-of-bounds-diagnostics.c

```diff
@@ -278,6 +278,23 @@ int *mallocRegion(void) {
   return mem;
 }
 
+int *custom_calloc(size_t a, size_t b) {
+  size_t res;
+  if (__builtin_mul_overflow(a, b, &res))
+    return 0;
+
+  return malloc(res);
+}
+
+int *mallocRegionOverflow(void) {
+  int *mem = (int*)custom_calloc(4, 10);
+
+  mem[20] = 10;
```

Comment:
Note: We don't have a FP here only because the `suppress-null-return-paths` heuristic is enabled by default - because of the state-split even in the concrete-concrete case.

Reply:
Thanks for note! I've just copy-pasted logic from tests above =)